test-scripts/src/main.rs

```diff
@@ -0,0 +1,725 @@
+    };
+
+    let mut client = sdk::Client::new(endpoint).await?;
+
+    // deploy module
+    let mod_ref = {
+        let module = WasmModule::from_file(&args.module)?;
+        let mod_ref = module.get_module_ref();
+        if client
+            .get_module_source(&mod_ref, BlockIdentifier::LastFinal)
+            .await
+            .is_ok()
+        {
+            eprintln!("Source module with reference {mod_ref} already exists.");
+        } else {
+            let nonce = client
+                .get_next_account_sequence_number(&admin.address)
+                .await?;
+            let tx = transactions::send::deploy_module(
+                &admin,
+                admin.address,
+                nonce.nonce,
+                TransactionTime::hours_after(1),
+                module,
+            );
+            let hash = client.send_account_transaction(tx).await?;
+            eprintln!("Send deployment transaction with hash {hash}.");
+            let (block_hash, result) = client.wait_until_finalized(&hash).await?;
+            if let Some(err) = result.is_rejected_account_transaction() {
+                anyhow::bail!("Failed to deploy module: {err:#?}");
+            }
+            eprintln!("Deployed module with reference {mod_ref} in block {block_hash}");
+        }
+        mod_ref
+    };
+
+    // initialize new instance
+    let (start_timestamp, end_timestamp, mut contract_client) = {
+        let start_timestamp = chrono::Utc::now()
+            .checked_add_signed(chrono::Duration::minutes(2))
+            .context("Time overflow.")?;
+        let election_start = start_timestamp.try_into()?;
+        let end_timestamp = chrono::Utc::now()
+            .checked_add_signed(chrono::Duration::minutes(args.election_duration + 2))
+            .context("Time overflow")?;
+
+        let election_end = end_timestamp.try_into()?;
+        let candidates = (0..args.num_options)
+            .map(|c| {
+                contract::ChecksumUrl {
+                    url:  format!("http://localhost:7000/{c}"),
+                    hash: contract::HashSha2256([0u8; 32]), // TODO
+                }
+            })
+            .collect();
+        let eligible_voters = contract::ChecksumUrl {
+            url:  "http://localhost:7000/voters".to_string(),
+            hash: contract::HashSha2256([1u8; 32]), // TODO
+        };
+        let init_param = contract::InitParameter {
+            admin_account: admin.address,
+            candidates,
+            guardians: guardians.iter().map(|g| g.address).collect(),
+            eligible_voters,
+            election_manifest: contract::ChecksumUrl {
+                url:  "http://localhost:7000/static/electionguard/election-manifest.json".into(),
+                hash: contract::HashSha2256(manifest_digest),
+            },
+            election_parameters: contract::ChecksumUrl {
+                url:  "http://localhost:7000/static/electionguard/election-parameters.json".into(),
```

Comment:
I think it would make sense to make the port configurable. Port 7000 is reserved for some OS process on macOS. I don't care what the default is so if you have a preference for 7000, by all means go with that, but it would be nice to not have to edit the script.

Reply:
Will add for the challenged OSs.

Reply:
Added base_url